Userland/Libraries/LibWeb/HTML/StructuredSerialize.cpp

```diff
@@ -148,13 +149,11 @@ class Serializer {
 
     WebIDL::ExceptionOr<void> serialize_string(Vector<u32>& vector, String const& string)
     {
-        u64 const size = string.code_points().length();
+        u64 const size = string.code_points().byte_length();
         // Append size of the string to the serialized structure.
         TRY_OR_THROW_OOM(m_vm, vector.try_append(bit_cast<u32*>(&size), 2));
-        for (auto code_point : string.code_points()) {
-            // Append each code point to the serialized structure.
-            TRY_OR_THROW_OOM(m_vm, vector.try_append(code_point));
-        }
+        // Append the bytes of the string to the serialized structure.
+        TRY_OR_THROW_OOM(m_vm, vector.try_append(bit_cast<u32*>(string.code_points().bytes()), AK::ceil_div(size, 4L)));
```

Comment:
Wait, won't this read past the end of the string if its length is not divisible by 4?

Reply:
Dangit yes, should probably construct this manually instead of relying on `bit_cast()` to do the job for me